File: Lab4/code/hw1.py
# ข้อ 1
Amount = float(input("จำนวนเงินสินค้าที่ซื้อ >> "))
Membership = input("คุณเป็นสมาชิกหรือไม่ >> ")
Discount = None
d = None

def funcDiscount(amount , percent):
    # https://brandinside.asia/how-to-calculate-percent-in-daily-life
    return amount * ((100 - percent) / 100)

if Amount > 5000 and (Membership == "Y" or Membership == "y"):
    d = 20
    Discount = funcDiscount(Amount , 20)
elif Amount > 5000 and (Membership == "N" or Membership == "n"):
    d = 10
    Discount = funcDiscount(Amount , 10)
else:
    d = 5
    Discount = funcDiscount(Amount , 5)
    
print(f"คุณซื้อสินค้าในราคา {Amount:,} บาท คุณได้ส่วนลด {d}% ราคาที่ต้องจ่ายจริงคือ {Discount:,.2f}")

File: Lab4/code/hw1.c
#include <stdio.h>

// ข้อ 1
float discount(float price , int percent){
    const float formulaOfProductDiscount = price * ((100 - percent)  / (float)100);
    return formulaOfProductDiscount;
}

void main(){
    float Amount , Discount;
    char Membership;

    puts("Amount of a product total price ?");
    scanf("%g" , &Amount);
    getchar();
    puts("Are you a membership ?");
    scanf("%c", &Membership);

    if(Amount > 5000 && Membership == 'Y' || Membership == 'y'){
        Discount = discount(Amount , 20);
    } else if(Amount > 5000 && Membership == 'N' || Membership == 'n'){
        Discount = discount(Amount , 10);
    } else {
        Discount = discount(Amount , 5);
    }

    printf("You pay the price %g bath" , Discount);
}